hie-base/Haskell/Ide/Engine/PluginTypes.hs

```diff
@@ -389,15 +426,53 @@ instance FromJSON (ParamVal 'PtFile) where
   parseJSON = withObject "file parameter object" $ \v -> ParamFile <$> v.: "file"
 
 instance FromJSON (ParamVal 'PtPos) where
```

Comment:
Using withObject here gives us better errors messages so it probably makes sense to leave it in or at least move it to the instance of Pos.